cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/CreateMoneyFlow.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.flow
+
+import co.paralleluniverse.fibers.Suspendable
+import com.example.contract.CashContract
+import com.example.state.CashState
+import com.example.utils.bankProvider
+import net.corda.core.contracts.Command
+import net.corda.core.contracts.requireThat
+import net.corda.core.flows.*
+import net.corda.core.identity.Party
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+
+object CreateMoneyFlow {
+    @InitiatingFlow
+    @StartableByRPC
+    class Initiator(val cashValue: Long, val moneyOwner: Party) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            val notary = serviceHub.networkMapCache.notaryIdentities[0]
+            val bank = serviceHub.bankProvider()
+
+            require(ourIdentity.owningKey == bank.owningKey) { "only bank can create money" }
+
+            val cashState = CashState(cashValue, bank, moneyOwner);
+            val cashCommand = Command(CashContract.Commands.Create(), bank.owningKey)
+
+            val txBuilder = TransactionBuilder(notary)
+                    .addOutputState(cashState, CashContract.ID)
```

Comment:
Can you make the flow a bit more versatile, and let it issue any number of states, with different owners?

Reply:
ok, let me try.